### PR TITLE
Update custom provider resource merging

### PR DESCRIPTION
**Pull request by pmuens**
This PRs implements the custom provider resource merging as described in https://github.com/serverless/serverless/issues/1832

Adding the functionality described in https://github.com/serverless/serverless/issues/1833 should be a "no-brainer" / "one-liner" as this PR already prepares everything.

Tested it manually with custom provider resources and the integration tests we have.

@flomotlik @eahefnawy would like to hear your feedback on this 👯

### CODE REVIEW

File: lib/plugins/aws/deploy/lib/initializeResources.js [hunk 1]
```diff
@@ -1,63 +1,27 @@
...
 const BbPromise = require('bluebird');
 
 module.exports = {
   initializeResources() {
-    const coreCFTemplate = this.serverless.utils.readFileSync(
+    // save the custom resources (e.g. Resources, Outputs etc.) the user has added
+    // so that they can be merged in later on
+    this.serverless.service.customProviderResources = {};
+
+    if (this.serverless.service.resources && this.serverless.service.resources.Resources) {
```
Comment (flomotlik) [inline lib/plugins/aws/deploy/lib/initializeResources.js]: I don't think the second check is necessary. if resources.Resources doesn't exist it doesn't really matter, but if people only have a description or Outputs they wouldn't be copied

Comment (flomotlik) [inline lib/plugins/aws/deploy/lib/initializeResources.js]: ah but that would be a change in functionality which we do in another PR. Alright

File: lib/plugins/aws/deploy/lib/initializeResources.js [hunk 2]
```diff
@@ -1,63 +1,27 @@
...
+      this.serverless.service.customProviderResources = this.serverless.service.resources;
```
Comment (flomotlik) [inline lib/plugins/aws/deploy/lib/initializeResources.js]: I think it would be better to not have another place to do custom resources. Otherwise plugins might write additional resources there and they don't get copied or copied at the wrong time.

Imho loading the core-cf file, deploying it to Cloudformation and then adding it to service.resources should all be done in the createStack.js file. If we try to do it here we have to do some workarounds that could lead to other issues, when we do it all in createStack.js we don't have any issues really

**Comment by pmuens** [comment]
@flomotlik updated the PR according to your comments.

The new merging makes lot more sense and is actually easier than the old approach...

File: lib/plugins/aws/deploy/lib/createStack.js [hunk 3]
```diff
@@ -9,14 +11,23 @@ module.exports = {
 
     const stackName = `${this.serverless.service.service}-${this.options.stage}`;
 
+    const coreCloudFormationTemplate = this.serverless.utils.readFileSync(
```
Comment (flomotlik) [inline lib/plugins/aws/deploy/lib/createStack.js]: Can you move this into a separate method so the code isn't here twice in the same file? Or is that not possible because of promises?

**Comment by pmuens** [comment]
@flomotlik updated the PR so that the CloudFormation template is only loaded once

File: lib/plugins/aws/deploy/lib/createStack.js (hunk 3 above)
Comment (pmuens) [inline lib/plugins/aws/deploy/lib/createStack.js]: Yes, will look into it 👍

**Comment by pmuens** [comment]
@flomotlik refactored the loading of the core CloudFormation template so that it's not repetitive
